app/test_main.py

```diff
@@ -2,0 +1,26 @@
+import pytest
+
+import app.main
+from app.main import can_access_google_page, has_internet_connection
+
+@pytest.mark.parametrize(
+    "init_url, mocked_valid_google_url, mocked_has_internet_connection, expected_result",
+    [
+        pytest.param("https://www.google.pl/",True, True, "Accessible", id="valid url and connection is True"),
+        pytest.param("https://www.google.pl/", True, False, "Not accessible", id="valid url is True and connection is False"),
+        pytest.param("https://www", False, True, "Not accessible", id="valid url is False and connection is True"),
+        pytest.param("https://www", False, False, "Not accessible", id="both are False"),
+    ]
+)
+def test_can_access_google_page(monkeypatch, init_url, mocked_valid_google_url, mocked_has_internet_connection, expected_result) -> None:
```

Comment:
Consider renaming the test function to be more descriptive, such as `test_can_access_google_page_with_various_conditions`, to better convey the purpose of the test according to the checklist guidelines.